packages/algo-ts/src/impl/primitives.ts

```diff
@@ -32,6 +32,7 @@ export function toExternalValue(val: uint64 | biguint | bytes | string) {
 }
 export const toBytes = (val: unknown): bytes => {
   if (val instanceof AlgoTsPrimitiveCls) return val.toBytes().asAlgoTs()
+  if ((val as BytesBacked).bytes) return BytesCls.fromCompat((val as BytesBacked).bytes).asAlgoTs()
```

Comment:
This isn't a very robust check as there's nothing stopping someone from creating a struct with a `bytes` property.

Reply:
:+1: It has been updated to `if (val instanceof ARC4Encoded) return val.bytes`